### PR TITLE
ENG-7627: Zinc crash

### PR DESCRIPTION
**Pull request by MatheusCavalca**
### Description
For https://elevatelabs.atlassian.net/browse/ENG-7627

The changes in this pull request are a potential fix for the long-standing Zinc iOS [crash](https://console.firebase.google.com/u/0/project/elevate-ios-be9c0/crashlytics/app/ios:com.elevateapp.elevate/issues/5602f96af5d3a7f76b43da30?time=last-seven-days&sessionEventKey=38b277a204d44c2aaad6e456ab829424_1742880622404319374). We noticed that when a Zinc bundle containing assets for a given game was not present in the file system, but was tracked by a locally generated file called `repo.json`, the app entered in an inconsistent state where it assumed that the assets were available and crashed when trying to locally load them. 

We couldn't reproduce this scenario, but we believe that's something that can happen because of race conditions when asynchronously downloading the many different games in parallel and updating the `repo.json`. Besides that, we believe that the same scenario can happen when the OS decides to clean up some random folders to free up some space. 

Even though we couldn't reproduce it organically, we were able to hit the same crash by manually deleting downloaded game assets from the file system, leaving the app in an inconsistent state where the `repo.json` file was marking the same game as available.

We were able to fix the crash by adding a validation to check if the assets are actually present in the file system before trying to load them. If not, the app recovers itself by re-downloading them. It's important to mention that the app only tries to download the game files that are actually missing – the fix doesn't cause additional downloads (other games). This is exactly what the Android app does in this scenario – we were able to confirm that by putting the Android app on the same state by using the exact same steps.

tl;dr we weren't able to organically reproduce the issue, but we were able to understand why it crashes in a certain way for a given scenario. We are (pretty) confident that this fix will solve most of the crashes (if not all of them). Also, we don't have any concerns that this fix will worsen the UX.

### CODE REVIEW

**Review by andrewcousins7 — APPROVED** [review]
LGTM

File: Zinc/Private/ZincRepoIndex.m [hunk 1]
```diff
@@ -199,22 +201,41 @@ - (void) setState:(ZincBundleState)state forBundle:(NSURL*)bundleResource
 
 - (ZincBundleState) stateForBundle:(NSURL*)bundleResource
 {
+    NSString* bundleID = bundleResource.zincBundleID;
+    ZincVersion bundleVersion = bundleResource.zincBundleVersion;
+
+    if (![self fileExistsForID:bundleID bundleVersion:bundleVersion]) {
+        return ZincBundleStateNone;
+    }
```
Comment (jerelevine) [inline Zinc/Private/ZincRepoIndex.m:209]: Could you add a test for this specific change? I'd like to make sure we're making Zinc more maintainable with any changes we make.

File: ZincTests/ZincRepoIndexTests.m [hunk 2]
```diff
@@ -22,22 +25,22 @@ - (void) _testDictionaryRoundtrip:(ZincRepoIndex*)index
 {
     NSError* error = nil;
     NSDictionary* dict = [index dictionaryRepresentation];
-    ZincRepoIndex* index2 = [ZincRepoIndex repoIndexFromDictionary:dict error:&error];
+    ZincRepoIndex* index2 = [ZincRepoIndex repoIndexFromDictionary:dict fileURL:self.url error:&error];
```
Comment (jerelevine) [inline ZincTests/ZincRepoIndexTests.m:28]: `self.url` appears to be `nil` for these tests. That seems like a mistake. Should we require `fileURL` to not be nil or should we handle the `nil` case in case it could happens in a released build?

**Review by jerelevine — APPROVED** [review]
LGTM! I left a couple of suggestions but nothing blocking testing out the fix

File: Zinc/Private/ZincRepoIndex.m (hunk 1 above)
Comment (MatheusCavalca) [reply Zinc/Private/ZincRepoIndex.m:209]: I've created a task to cover these changes with unit tests: https://elevatelabs.atlassian.net/browse/ENG-7751
